eng/pipelines/templates/steps/analyze.yml

```diff
@@ -4,13 +4,22 @@ parameters:
   TestPipeline: false
 
 steps:
+  - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
+    parameters:
+      ServiceDirectory: ${{parameters.ServiceDirectory}}
+
+  - template: /eng/pipelines/templates/steps/set-artifact-packages.yml
+    parameters:
+      PackageInfo: $(Build.ArtifactStagingDirectory)/PackageInfo
+      Artifacts: ${{ parameters.Artifacts }}
+
   - template: /eng/common/pipelines/templates/steps/check-spelling.yml
 
   - task: PowerShell@2
     inputs:
       targetType: 'filePath'
-      filePath: eng/scripts/spell-check-public-api.ps1
-      arguments: -ServiceDirectory ${{ parameters.ServiceDirectory }}
+      filePath: eng/scripts/spell-check-public-apis.ps1
+      arguments: -ChangedServices "$(ChangedServices)"
```

Comment:
I know this was already there but I wonder if this needed given that the check-spelling.yml already checks spelling on any changed files. Perhaps this is interesting on the internal pipelines running against a given service directory though.

Reply:
Yeah, I was basically going for exact 1:1 replacement, just handling dynamic packages.